editor/plugins/theme_editor_plugin.cpp

```diff
@@ -1577,7 +1578,7 @@ void ThemeItemEditorDialog::_remove_data_type_items(Theme::DataType p_data_type,
 	Ref<Theme> old_snapshot = edited_theme->duplicate();
 	Ref<Theme> new_snapshot = edited_theme->duplicate();
 
-	UndoRedo *ur = EditorNode::get_singleton()->get_undo_redo();
+	Ref<EditorUndoRedoManager> ur = EditorNode::get_undo_redo();
```

Comment:
I think most of the temporary objects in this file can be replaced by `Ref<EditorUndoRedoManager> &ur = ...`

Using a reference avoids repeatedly increasing & decreasing the refcounter in `RefCounted`.